packages/dds/tree/src/util/utils.ts

```diff
@@ -12,3 +12,22 @@ export function clone<T>(original: T): T {
 export function fail(message: string): never {
     throw new Error(message);
 }
+
+/**
+ * Used as a default branch in switch statements to enforce that all possible branches are accounted for.
```

Comment:
Clarify this is a type system thing.

```suggestion
 * Use as a default branch in switch statements to enforce (at compile time) that all possible branches are accounted for,
  * according to the TypeScript type system.
  * As an additional protection, errors if called.
```